src/Broker/Broker.c

```diff
@@ -26,28 +26,40 @@
 #include <stdint.h>
 
 #include <Packetizer.h>
+#include <ThinPacketizer.h>
+#include <DriverHelper.h>
 
-static Packetizer packetizer_data = { 0 };
+static Packetizer packetizers_data[SYSTEM_BUSES_NUMBER] = { 0 };
 
 __attribute__((section(".sdramMemorySection"))) static uint8_t packetizer_buffer[BROKER_BUFFER_SIZE];
 extern driver_send_function bus_to_driver_send_function[SYSTEM_BUSES_NUMBER];
 extern void* bus_to_driver_private_data[SYSTEM_BUSES_NUMBER];
+extern enum PacketizerCfg bus_to_packetizer_cfg[SYSTEM_BUSES_NUMBER];
 extern deliver_function interface_to_deliver_function[INTERFACE_MAX_ID];
+extern PacketizerFunctions packetizers[PACKETIZER_MAX_ID];
 
 extern void Broker_acquire_lock();
 extern void Broker_release_lock();
 
 void
-Broker_initialize()
+Broker_initialize(enum SystemBus valid_buses[SYSTEM_BUSES_NUMBER])
 {
-    Packetizer_init(&packetizer_data);
+    for(int i = 0; i < SYSTEM_BUSES_NUMBER; ++i) {
+        enum SystemBus bus_id = valid_buses[i];
+        if (bus_id == BUS_INVALID_ID) {
+            break;
+        }
+        
+        enum PacketizerCfg packetizer_name = bus_to_packetizer_cfg[bus_id];
```

Comment:
Name implies a string...

Reply:
After reading a bit more of the code, I think that we should think carefully about the names, because otherwise, we will be lost in the indirections.

packetizers[] hold function pointers, indexed by packetizer_type -> it recommend renaming it to packetizers_functions
packerizers_data[] hold packetizer instance data, indexed by bus_id.

Reply:
done